ert_shared/socket_resolver.py

```diff
@@ -0,0 +1,49 @@
+import sys
+import socket
+
+
+class SocketResolver:
+    def __init__(self):
+        self._range = range(51820, 51840 + 1)
+        self._used = {}
+
+    def assign_socket(self, user: str, host: str, port: int = None) -> socket.socket:
```

Comment:
"user" is a bit overloaded and could mean username. Should we maybe have the following enum, which is what the users of this module would call with?
```py
class Service(Enum):
    storage = 0
    evaluator = 1
```

Ie, `socket_resolver.get_port(Service.storage)`

Reply:
totally agree, but just throwed it into the code for discussion. I haven't resolved the old lookup code in prefect-ensemble either - which i'm not sure how it works..  But yes - i might think that different parts of our code using a port could be represented as an enum 👍   Just feel free to suggest a totally different approach as well, or things which might be stupid with my approach. I currently digging into the prefect ensemble code where the lookup and close of a socket is performed, not quite sure how it works so maybe need some input on that.